packages/@coorpacademy-components/src/molecule/learning-priority-modal/index.js

```diff
@@ -0,0 +1,253 @@
+  const {active, filter, itemTotal, onClick} = props;
+  const {skin} = context;
+  const primarySkinColor = get('common.primary', skin);
+
+  const Content = useCallback(
+    () => (
+      <div>
+        {filter}
+        <span
+          className={active ? style.skillFilterNumber : style.skillFilterNumberInActive}
+          style={{
+            backgroundColor: active ? convert(`color(${primarySkinColor} a(0.07))`) : '#EAEAEB',
+            color: active ? primarySkinColor : '#515161'
+          }}
+        >
+          {itemTotal}
+        </span>
+      </div>
+    ),
+    [filter, itemTotal, active, primarySkinColor]
+  );
+
+  const buttonProps = {
+    customStyle: {
+      backgroundColor: active ? convert(`color(${primarySkinColor} a(0.07))`) : 'transparent',
+      color: active ? primarySkinColor : '#9999A8',
+      transition: 'background-color 0.15s ease-in-out, color 0.15s ease-in-out',
+      width: 'fit-content'
+    },
+    onClick,
+    content: <Content />,
+    'data-name': 'change-skill-focus-button'
+  };
+
+  return <ButtonLink {...buttonProps} />;
+};
+
+FilterButton.contextTypes = {
+  skin: Provider.childContextTypes.skin,
+  translate: Provider.childContextTypes.translate
+};
+
+FilterButton.propTypes = {
+  active: PropTypes.bool,
+  filter: PropTypes.string,
+  itemTotal: PropTypes.number,
+  onClick: PropTypes.func
+};
+
+const LearningPriorityModal = (props, context) => {
+  const {
+    priorities,
+    selectedPriorities,
+    isOpen,
+    isLoading,
+    filters: {options, onChange},
+    onCancel,
+    onAdd,
+    onClose
+  } = props;
+  const {translate} = context;
+
+  const [selectedPriority, setSelectedPriority] = useState('');
+  const [searchValue, setSearchValue] = useState('');
+  const [searchResults, setSearchResults] = useState(priorities);
+
+  const handleCancel = useCallback(() => {
+    setSelectedPriority('');
+    onCancel();
+  }, [setSelectedPriority, onCancel]);
+
+  const handleClose = useCallback(() => {
+    setSelectedPriority('');
+    onClose();
+  }, [setSelectedPriority, onClose]);
+
+  const handleSearch = useCallback(
+    value => {
+      setSearchValue(value);
+      setSearchResults(priorities.filter(priority => searchValueIncluded(priority.title, value)));
+    },
+    [priorities, setSearchValue, setSearchResults]
+  );
+
+  const handleSearchReset = useCallback(() => {
+    setSearchValue('');
+    setSearchResults(priorities);
+  }, [priorities, setSearchValue, setSearchResults]);
+
+  useEffect(() => {
+    setSearchResults(priorities);
+  }, [priorities]);
+
+  const priorityList = useMemo(() => {
+    return searchResults.map(priority => {
+      const isSelectedPriority = selectedPriorities.some(selected => selected === priority.ref);
+      return {
+        title: priority.title,
+        ref: priority.ref,
+        type: priority.type,
+        courses: priority.courses,
+        selected: isSelectedPriority,
+        disabled: isSelectedPriority
```

Comment:
❗ can't be both 😅 
Also would recommend a `assign({selected: isSelectedPriority, disabled: !isSelectedPriority}, pick(['title', 'ref, 'courses', 'type'], priority)`
(maybe, the pick if no neccessy if it's no problem having extra properties)

Reply:
This is for list of priority that have already been selected before. So they should be disabled also. I've update the coding convention for this

Reply:
👌 , sorry for that misunderstanding